arviz/plots/pairplot.py

```diff
@@ -128,7 +128,7 @@ def plot_pair(data, var_names=None, coords=None, figsize=None, textsize=None, ki
             for j in range(i, numvars - 1):
                 var2 = _posterior[j + 1]
 
-                ax = plt.subplot(gs[j, i])
+                ax = fig.add_subplot(gs[j, i])
```

Comment:
Since the shape of the subplot is known (numvars-1, numvars-1), why not just create `fig, ax = plt.subplots(numvars-1, numvars-1, figsize=figsize)` above and index to the right one here?

Reply:
It'll then create all the other empty plots in the upper triangular

Reply:
Is this implementation what you are referring to?
https://github.com/canyon289/arviz/tree/alternative_pairplot

Example image
https://user-images.githubusercontent.com/7213793/45938966-bb5f3500-bf83-11e8-9461-eda0054c3539.png

Reply:
I see, agree with you - thanks for the explanation!